specs/predeploys.md

```diff
@@ -93,290 +118,176 @@ In the Bedrock system, this contract will no longer be used as part of the
-    function approve(address spender, uint256 amount) external returns (bool);
-    function transferFrom(
-        address from,
-        address to,
-        uint256 amount
-    ) external returns (bool);
-}
-```
+Address: `0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000`
 
-## WETH9
+The `LegacyERC20ETH` predeploy represents all ether in the system before the
+Bedrock upgrade. All ETH was represented as an ERC20 token and users could opt
+into the ERC20 interface or the native ETH interface.
 
-`WETH9` is the standard implementation of Wrapped Ether on Optimism.
+The upgrade to Bedrock migrates all ether out of this contract and moves it to
+its native representation. All of the stateful methods in this contract will
+revert after the Bedrock upgrade.
 
-```solidity
-interface WETH9 {
-    function name() public returns (string);
-    function symbol() public returns (string);
-    function decimals public returns (uint8);
+This contract is deprecated and its usage should be avoided.
 
-    event  Approval(address indexed src, address indexed guy, uint wad);
-    event  Transfer(address indexed src, address indexed dst, uint wad);
-    event  Deposit(address indexed dst, uint wad);
-    event  Withdrawal(address indexed src, uint wad);
+## WETH9
 
-    function balanceOf(address) public returns (uint);
-    function allowance(address, address) public returns (uint);
+[Implementation](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/vendor/WETH9.sol)
 
-    function deposit() public;
-    function withdraw(uint wad) public;
-    function totalSupply() public view returns (uint);
-    function approve(address guy, uint wad) public returns (bool);
-    function transfer(address dst, uint wad) public returns (bool);
+Address: `0x4200000000000000000000000000000000000006`
 
-    function transferFrom(
-        address src,
-        address dst,
-        uint wad
-    ) public returns (bool);
-}
-```
+`WETH9` is the standard implementation of Wrapped Ether on Optimism. It is a
+commonly used contract and is placed as a predeploy so that it is at a
+deterministic address across chains.
 
 ## L2CrossDomainMessenger
 
-The `L2CrossDomainMessenger` is used to give a better user experience when
-sending cross domain messages from L2 to L1. It extends the
-`CrossDomainMessenger`, which allows for replayability of messages. Any calls
-through the `l1CrossDomainMessenger` go through the `L2CrossDomainMessenger`.
+[Implementation](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/L2/L2CrossDomainMessenger.sol)
+
+Address: `0x4200000000000000000000000000000000000007`
+
+The `L2CrossDomainMessenger` gives a higher level API for sending cross domain
+messages. It maintains a mapping of L1 messages that have been relayed to L2
+to prevent replay attacks and also allows for replayability if the L1 to L2
+transaction reverts on L2.
```

Comment:
Higher-level compared to what? (OptimismPortal I presume)

Esp. confusing because below you say "The `L2StandardBridge` is a higher level API built on top of the
`L2CrossDomainMessenger`" so people will wonder if either that text or the text above this comment was not mistaken.

Reply:
Ok good point, will clarify this section